versioning/versioning.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+GIT_VERSION_COMMAND='git describe --tags --abbrev=8 --dirty 2> /dev/null'
+# Based on https://semver.org/#semantic-versioning-200 but allows `v` in front to match git tags
+GIT_SEMVER_REGEX=/^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
+
+class Versioning
+  class << self
+    def verify_git!
+      unless git_exists?
+        raise('The command `git` does not exist!')
+      end
+
+      unless git_dir?
+        raise("The current directory `#{Dir.pwd}` is not a git tree!")
+      end
+    end
+
+    def verify_semver_tags!
+      unless `#{GIT_VERSION_COMMAND}`.strip =~ GIT_SEMVER_REGEX
+        raise('A git tag with an semantic version is required!')
+      end
+    end
+
+    def current_version
+      version=`#{GIT_VERSION_COMMAND}`.strip
+      version.gsub(/^v?(.*)$/, '\1')
```

Comment:
I think we require ruby-2.5 or later anyways, in which case I find it more readable to write:
```suggestion
      version.delete_prefix("v")
```